kernel/src/main/java/org/kframework/unparser/ToJson.java

```diff
@@ -137,7 +139,32 @@ public static JsonStructure toJson(Att att) {
 
         JsonObjectBuilder jattKeys = Json.createObjectBuilder();
         for (Tuple2<String,String> key: JavaConverters.seqAsJavaList(att.att().keys().toSeq())) {
-            jattKeys.add(key._1(), att.att().get(key).get().toString());
+            if (key._1().equals(Location.class.getName())) {
+                JsonArrayBuilder locarr = Json.createArrayBuilder();
+                Location loc = att.get(Location.class);
+                locarr.add(loc.startLine());
+                locarr.add(loc.startColumn());
+                locarr.add(loc.endLine());
+                locarr.add(loc.endColumn());
+                jattKeys.add(key._1(), locarr.build());
+            } else if (key._1().equals(Source.class.getName())){
+                jattKeys.add(key._1(), att.get(Source.class).source());
+            } else if (key._1().equals(Production.class.getName())){
+                jattKeys.add(key._1(), toJson(att.get(Production.class)));
+            } else if (key._1().equals(Sort.class.getName())){
+                jattKeys.add(key._1(), toJson(att.get(Sort.class)));
+            } else if (key._1().equals("bracketLabel")) {
+                jattKeys.add(key._1(), toJson(att.get("bracketLabel", KLabel.class)));
+            } else if (key._1().equals(Att.PREDICATE())) {
+                jattKeys.add(key._1(), toJson(att.get(Att.PREDICATE(), Sort.class)));
+            } else if (key._1().equals("cellOptAbsent")) {
+                jattKeys.add(key._1(), toJson(att.get("cellOptAbsent", Sort.class)));
+            } else if (key._1().equals("cellFragment")) {
+                jattKeys.add(key._1(), toJson(att.get("cellFragment", Sort.class)));
+            } else if (key._1().equals("sortParams")) {
+                jattKeys.add(key._1(), toJson(att.get("sortParams", Sort.class)));
+            } else
+                jattKeys.add(key._1(), att.att().get(key).get().toString());
```

Comment:
Likewise here, if the logic is similar to the other if-else chain, how do we ensure they stay in sync?

Reply:
This is the serialization logic. The pair of the above^.